integration-tests/client/chainlink_config_builder.go

```diff
@@ -0,0 +1,129 @@
+package client
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/pelletier/go-toml/v2"
+	"github.com/rs/zerolog/log"
+	"github.com/smartcontractkit/chainlink-testing-framework/blockchain"
+	"github.com/smartcontractkit/chainlink/core/assets"
+	v2 "github.com/smartcontractkit/chainlink/core/chains/evm/config/v2"
+	chainlinkConfig "github.com/smartcontractkit/chainlink/core/services/chainlink"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+type ChainlinkConfigBuilder struct {
+	config *chainlinkConfig.Config `toml:"-"`
+}
+
+// NewDefaultConfig gets the base config for
+func NewDefaultConfig() *ChainlinkConfigBuilder {
+	conf := &chainlinkConfig.Config{}
+	conf.RootDir = Pointer("./clroot")
+	conf.DevMode = false
+
+	conf.Log.JSONConsole = Pointer(true)
+
+	conf.WebServer.AllowOrigins = Pointer("*")
+	conf.WebServer.SecureCookies = Pointer(false)
+	conf.WebServer.SessionTimeout = models.MustNewDuration(time.Hour * 999)
+	conf.WebServer.TLS.HTTPSPort = Pointer(uint16(0))
+	return &ChainlinkConfigBuilder{conf}
+}
+
+// AddNetworks adds networks to the config, enabling forwarders for each network if specified
+func (c *ChainlinkConfigBuilder) AddNetworks(
+	enableForwarders bool,
+	networks ...*blockchain.EVMNetwork,
+) *ChainlinkConfigBuilder {
+	for _, network := range networks {
+		nodes := []*v2.Node{}
+		for nodeId := range network.URLs {
+			wsURL, err := models.ParseURL(network.URLs[nodeId])
+			if err != nil {
+				log.Fatal().Str("URL", network.URLs[nodeId]).Err(err).Msg("Error in URL formatting")
+			}
+			httpURL, err := models.ParseURL(network.HTTPURLs[nodeId])
+			if err != nil {
+				log.Fatal().Str("URL", network.HTTPURLs[nodeId]).Err(err).Msg("Error in URL formatting")
+			}
+			nodes = append(nodes, &v2.Node{
+				Name:    Pointer(fmt.Sprintf("node-%d", nodeId)),
+				WSURL:   wsURL,
+				HTTPURL: httpURL,
+			})
+		}
+
+		c.config.EVM = append(c.config.EVM, &v2.EVMConfig{
+			ChainID: utils.NewBigI(network.ChainID),
+			Enabled: Pointer(true),
+			Nodes:   nodes,
+			Chain: v2.Chain{
```

Comment:
how about passing Chain as parameter and use a default for usual cases? CCIP tests would need some customizations here.